### PR TITLE
Update documentation for BaseEstimator and BaseSampler: issues #8517 and #8516

### LINKED ISSUE #8516: Example in "Overview of Estimator" is missing an import statement
### Environment

- **Qiskit Terra version**: 0.37.1

Cutting and pasting the example given in "Overview of Estimator" almost works. But, `Estimator` is not imported.
This is the [API doc for Primitives](https://qiskit.org/documentation/apidoc/primitives.html), not in the docstring for the `Estimator` class.

Adding
```python
from qiskit.primitives import Estimator
```

to this list

https://github.com/Qiskit/qiskit-terra/blob/c318725d679b0eb3aa8874fb9755fa0567c5e94b/qiskit/primitives/base_estimator.py#L61-L62

does allow the example to run.

The same problem occurs for `Sampler`. Is this intentional? Maybe the idea is that in an example in a description of a class, the reader is expected to copy the example code and then add a line to import the class being described ?

Whether it's intentional or not, I think adding a single line to these examples would make them more convenient to use.

### What is happening?

The example for `Estimator` fails because of a missing import statement.

### How can we reproduce the issue?

Copy and paste the example into a Python session.

### What should happen?

The example should run and print the output.

### Any suggestions?

Add a line in the example to import the class.

### LINKED ISSUE #8517: Documentation for inputs to an Estimator instance disagree with implementation
### Environment

- **Qiskit Terra version**:  0.37.1
 

### What is happening?

The documentation in [BaseEstimator](https://github.com/Qiskit/qiskit-terra/blob/c318725d679b0eb3aa8874fb9755fa0567c5e94b/qiskit/primitives/base_estimator.py#L114) says that when calling an instance four arguments are supplied. The third and fourth are parameters and parameter values.

But the implementation and the example are different. They take only three arguments. Parameter values are passed, but not parameters. Instead, in the call method, the parameters are retrieved from the circuit.

The above is in the file for `BaseEstimator`, but the doc string and example contained therein refer to `Estimator`. It appears online in the API reference section.
 

### How can we reproduce the issue?

Read the source

### What should happen?

Documented interface should agree with implementation.

### Any suggestions?

_No response_

### PR DESCRIPTION
**Pull request by sebastian-mair**
- [no] I have added the tests to cover my changes.
- [yes] I have updated the documentation accordingly.
- [yes] I have read the CONTRIBUTING document.

Linked Issues: 
* Fix #8516
* Fix #8517

### Summary
I updated the sample usages for qiskit.primitives.BaseEstimator and qiskit.primitives.BaseSampler, as well as a brief adaptation  of the general description of the class usage.
The updated description matches the current implementation - the existing documentation described a deprecated case and was adapted.

### Details and comments
This is my first contribution to any open source project, so hello there ;)

### CODE REVIEW

**Comment by HuangJunye** [comment]
@sebastian-mair Thanks a lot for making this contribution. As this is your first time contributing in Qiskit Terra, please read the [Contributing to Qiskit page](https://qiskit.org/documentation/contributing_to_qiskit.html), especially the [PR checklist](https://qiskit.org/documentation/contributing_to_qiskit.html#pull-request-checklist).

**Comment by HuangJunye** [comment]
@ikkoham Would you like to review this PR?

**Review by ikkoham — CHANGES_REQUESTED** [review]
Congratulation on your great first contribution to OSS!
Thanks to you the documentation is much better.

It almost looks good to me. I'd like to left some minor comments.

File: qiskit/primitives/base_estimator.py [hunk 1]
```diff
@@ -19,48 +19,39 @@
 
 Estimator class estimates expectation values of quantum circuits and observables.
 
-An estimator object is initialized with multiple quantum circuits and observables
-and users can specify pairs of quantum circuits and observables
-to estimate the expectation values.
-
-An estimator is initialized with the following elements.
+An estimator is initialized with an empty parameter set. The estimator is used to
+create a :class:`qiskit.primitives.PrimitiveJob`, via the
```
[the remark this answers is not in the document]
Comment (sebastian-mair) [reply qiskit/primitives/base_estimator.py:23]: Thank you, I'll correct that.

**Review by t-imamichi — DISMISSED** [review]

**Comment by ikkoham** [comment]
@t-imamichi I'll update this PR according to my comments.

**Comment by ikkoham** [comment]
I checked the generated documents.
image
image

**Review by ikkoham — APPROVED** [review]
LGTM. Thank you for your contribution to qiskit.

**Review by t-imamichi — APPROVED** [review]

**Pull request merged**